### ISSUE TITLE
Reproducing Paper Scores

### ISSUE THREAD

**Issue by dominik-zietlow** [rusty1s/deep-graph-matching-consensus#1]
Hi guys!

Would you mind providing example files that reproduce the scores from the paper?
The example files are not made for this, are they? The resulting scores are terrible :-P

Many thanks!
Dominik

**Comment by rusty1s** [comment]
They actually are. It is hard to tell what may cause this behaviour without further details on your part. Which example produces which result with which configuration? Are you on PyG master?

**Comment by dominik-zietlow** [comment]
Thanks for your quick reply! Sorry I didn't spend much time on figuring out the potential problem as I assumed that they are really just random examples that run some code ;-)

I was running example/pascal.py, the accuracies are not that promising:
https://imgur.com/a/8Rddrnw

torch-geometric installed from master - I might miss something very obvious here :-D

**Comment by rusty1s** [comment]
Can you also post your PyTorch version? I will try to reproduce this behavior :)

**Comment by dominik-zietlow** [comment]
That was run with PyTorch 1.4.0, thanks for your efforts!

**Comment by dominik-zietlow** [comment]
attrs==19.3.0
bayesian-optimization==1.0.1
blossom-v==0.0.0
certifi==2019.11.28
chardet==3.0.4
Click==7.0
cloudpickle==1.2.2
cluster==1.2
cma==2.7.0
colorama==0.4.3
cycler==0.10.0
decorator==4.4.1
easydict==1.9
filelock==3.0.12
funcsigs==1.0.2
gitdb2==2.0.6
GitPython==3.0.5
googledrivedownloader==0.4
GPUtil==1.4.0
graph-matching-py==0.0.1
h5py==2.10.0
idna==2.8
imageio==2.6.1
importlib-metadata==1.5.0
isodate==0.6.0
joblib==0.14.1
jsonschema==3.2.0
kiwisolver==1.1.0
llvmlite==0.31.0
matplotlib==3.1.2
more-itertools==8.2.0
networkx==2.4
nevergrad==0.3.1
numba==0.48.0
numpy==1.18.1
packaging==20.1
pandas==1.0.0
pathlib2==2.3.5
Pillow==6.2.2
pluggy==0.13.1
plyfile==0.7.1
protobuf==3.11.2
py==1.8.1
pykeops==1.3
pyparsing==2.4.6
pyrsistent==0.15.7
pytest==5.3.5
python-dateutil==2.8.1
pytz==2019.3
pyuv==1.4.0
PyWavelets==1.1.1
PyYAML==5.3
ray==0.8.1
rdflib==4.2.2
redis==3.4.0
requests==2.22.0
scikit-image==0.16.2
scikit-learn==0.22.1
scipy==1.4.1
seaborn==0.10.0
six==1.14.0
sklearn==0.0
smmap2==2.0.5
tensorboardX==2.0
torch==1.4.0
torch-cluster==1.4.5
torch-geometric==1.3.2
torch-scatter==1.4.0
torch-sparse==0.4.3
torch-spline-conv==1.1.1
torchvision==0.5.0
tqdm==4.42.0
typing-extensions==3.7.4.1
urllib3==1.25.8
wcwidth==0.1.8
zipp==2.1.0

**Comment by rusty1s** [comment]
I just checked the scripts on two independent computers, and can reach the expected results with both using PyTorch 1.4.0. There can be multiple issues you can check to verify your installation/data:
* Please run other scripts as well, *e.g.* `pascal_pf.py` and see if the problem remains.
* If this script works for you, try to delete the processed files from `PascalVOC` in `data/PascalVOC/processed` and run the script again. There may be files in use from an older PyG version.
* If this script does not work for you, try to check if you have valid `torch-scatter` and `torch-spline-conv` installations by running the test suites of each package, or by running `examples/cora.py` from the PyTorch Geometric repository (should reach about 88-90%).

I hope this helps! Please let me know :)

**Comment by dominik-zietlow** [comment]
First of all thanks for the response! 

For stupid reasons it was non trivial to compile stuff (during torch-geometric installation) on the machine on which the code had to be executed, so it was compiled on a very similar one. This seems to have caused the trouble (leading to NaNs in the loss for pascal_pf and to ridiculously poor performance on pascal - which is somewhat terrifying to see that things may run without crashing but misbehaving because of that...). 

I can now run pascal_pf with good scores, the pascal.py script is still preprocessing but I am confident it will work :)

Again thanks for your support!

**Comment by rusty1s** [comment]
Glad to hear :) I will try to improve the speed of pre-processing `PascalVOC`.
